src/utils/uid.ts

```diff
@@ -0,0 +1,23 @@
+import { getCurrentInstance } from 'vue';
+
+import type { ComponentInternalInstance } from 'vue';
+
+let _uid = 0;
+const _map = new WeakMap<ComponentInternalInstance, number>();
+
+/**
+ * Returns an unique id for current component instance
+ * Inspired by Vuetify: https://github.com/vuetifyjs/vuetify/blob/master/packages/vuetify/src/util/getCurrentInstance.ts#L26
+ * @returns An UID
+ */
+export function getUid(): number {
+  const vm = getCurrentInstance();
```

Comment:
Maybe we could give this variable a more comprehensible name (e.g `currentInstance`)